File: engine/model/src/main/scala/net/server/channel/packet/UseTeleportRockPacket.scala
package net.server.channel.packet

class UseTeleportRockPacket(private var _position: Short, private var _itemId: Int, private var _vip: Boolean, private var _mapId: Int, private var _name: String) extends AbstractUseCashItemPacket(_position, _itemId) {
  def vip: Boolean = _vip

  def mapId: Int = _mapId

  def name: String = _name
}

File: engine/engine-model/src/main/java/net/server/channel/packet/cash/use/UseTeleportRockPacket.java
package net.server.channel.packet.cash.use;

public class UseTeleportRockPacket extends AbstractUseCashItemPacket {
   private final Boolean vip;

   private final Integer mapId;

   private final String name;

   public UseTeleportRockPacket(Short position, Integer itemId, Boolean vip, Integer mapId, String name) {
      super(position, itemId);
      this.vip = vip;
      this.mapId = mapId;
      this.name = name;
   }

   public Boolean vip() {
      return vip;
   }

   public Integer mapId() {
      return mapId;
   }

   public String name() {
      return name;
   }
}
